integration/docker_engine_test.go

```diff
@@ -0,0 +1,73 @@
+// +build integration
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aquasecurity/trivy/internal"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRun_WithDockerEngine(t *testing.T) {
+	testCases := []struct {
+		name               string
+		testfile           string
+		expectedOutputFile string
+		expectedError      error
+	}{
+		{
+			name:               "happy path, alpine:3.10",
+			expectedOutputFile: "testdata/alpine-310.json.golden",
+			testfile:           "testdata/fixtures/alpine-310.tar.gz",
+		},
+	}
+
+	for _, tc := range testCases {
+		ctx := context.Background()
+		testfile, _ := os.Open(tc.testfile)
+		cli, _ := client.NewClientWithOpts(client.FromEnv)
+
+		// ensure image doesnt already exists
+		_, _ = cli.ImageRemove(ctx, tc.testfile, types.ImageRemoveOptions{
```

Comment:
ditto

Reply:
Actually this is an error we should ignore. This is a best effort check to remove the image if it already exists for some reason (cruft). In a happy case where we get a fresh start without cruft, the error will not be nil. That's why I'm ignoring it.

Reply:
I understand.